presidio-image-redactor/presidio_image_redactor/image_processing_engine.py

```diff
@@ -0,0 +1,383 @@
+import numpy as np
+import PIL
+from PIL import Image, PngImagePlugin
+import cv2
+
+from typing import Union, Tuple, Optional
+
+
+class ImagePreprocessor:
+    """ImagePreprocessor class.
+
+    Parent class for image preprocessing objects.
+    """
+
+    def __init__(self, use_greyscale: bool = True) -> None:
+        """Initialize the ImagePreprocessor class.
+
+        :param use_greyscale: Whether to convert the image to greyscale.
+        """
+        self.use_greyscale = use_greyscale
+
+    def preprocess_image(
+        self, image: PngImagePlugin.PngImageFile
```

Comment:
I believe we should be able to just use PIL's standard `Image` type.

@ayabel - Would you mind replacing all `PngImagePlugin.PngImageFile` with just `Image.Image` and seeing if that works? If it does, please update the docstrings to indicate PIL image type rather than the specific PNG type.